### PR TITLE
46832_view_users_with_filter

### PR DESCRIPTION
**Pull request by zenith19**

### CODE REVIEW

**Comment by rofiuddin** [comment]
ok

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all.paginate(:page => params[:page], 
+        :per_page => 15)
+    if params[:supervisor] == '1'
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: list all supervisor at server side
please use other controller with namespace supervisor

File: app/models/user.rb [hunk 2]
```diff
@@ -1,6 +1,14 @@
 class User < ActiveRecord::Base
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
+  has_many :activities
+  has_many :courses, through: :users_course
+  has_many :subjects, through: :users_subject
+  has_many :tasks, through: :users_tasks
+
+  scope :find_all_supervisors, -> { where(supervisor: true) }
```
Comment (chuanhtuan) [inline app/models/user.rb]: scope :supervisors, -> { where(supervisor: true) }

File: app/models/user.rb [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+  scope :find_all_trainees, -> { where(supervisor: false) }
```
Comment (chuanhtuan) [inline app/models/user.rb]: scope :trainees, -> { where(supervisor: false) }

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all.paginate(:page => params[:page],
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: =>

File: .gitignore [hunk 5]
```diff
@@ -22,4 +22,5 @@ pickle-email-*.html
 config/settings.local.yml
 config/settings/*.local.yml
 config/environments/*.local.yml
-Gemfile.lock
+/Gemfile.lock
+/db/schema.rb
```
Comment (chuanhtuan) [inline .gitignore]: db/schema.rb must be committed

File: app/controllers/supervisor/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class Supervisor::UsersController < ApplicationController
+  def index
+    @users = User.all.paginate(:page => params[:page], per_page: 15)
```
Comment (chuanhtuan) [inline app/controllers/supervisor/users_controller.rb]: =>

File: app/controllers/supervisor/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+    if params[:supervisor] == '1'
```
Comment (chuanhtuan) [inline app/controllers/supervisor/users_controller.rb]: '1' must be defined as a constants in user model

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.trainees.paginate(page: params[:page], per_page: 15)
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ()

File: config/routes.rb [hunk 9]
```diff
@@ -1,9 +1,16 @@
...
   devise_for :users, controllers: { registrations: "registrations" }
+  
   root             "static_pages#home"
   get "home"    => "static_pages#home"
   get "about"   => "static_pages#about"
   get "contact" => "static_pages#contact"
-
+  
   resources :subjects
+  get 'users/index'
```
Comment (chuanhtuan) [inline config/routes.rb]: resource :user, only: [index]

File: config/routes.rb [hunk 10]
```diff
@@ -1,9 +1,16 @@
...
+  
+  namespace :supervisor do
+    get 'users/index'
```
Comment (chuanhtuan) [inline config/routes.rb]: resource :user, only: [index]

**Pull request merged**